net: lwm2m: Fix FOTA Pull firmware transfer when Package URI is empty

marinjurjevic opened the pull request:
Fix for a problem in current lwm2m firmware object implementation. If empty string is sent to Package URI while object's state machine is already in STATE_IDLE, object will reset State and Update Result and begin transfer. From OMA specification:

> Writing an empty string to Package URI Resource or setting the Package Resource to NULL (‘\0’), resets the Firmware Update State Machine: the State Resource value is set to Idle and the Update Result Resource value is set to 0.

Reset part is fine, but transfer should not begin when an empty string is received (it's a special case for resetting state machine).

rlubos commented on subsys/net/lib/lwm2m/lwm2m_obj_firmware.c:
```diff
@@ -247,7 +247,12 @@ static int package_uri_write_cb(uint16_t obj_inst_id, uint16_t res_id,
+		{
+			lwm2m_firmware_start_transfer(package_uri);
+		}
+
```
No need for this empty line.

rlubos reviewed (changes requested):
Please squash the commits and follow the [commit guidelines](https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines) regarding the commit message format (use `net: lwm2m: ` prefix).

I think that eventually we should allow to reset the state machine also in other states than IDLE, but it seems like a larger update, so the change in this PR looks good for now.

marinjurjevic committed:
net: lwm2m: Fix FOTA Pull firmware transfer when Package URI is empty

Fix for a problem in current lwm2m firmware object implementation.
Transfer should not begin when an empty string is received.

Signed-off-by: Marin Jurjevic <marin.jurjevic@hotmail.com>

marinjurjevic commented on subsys/net/lib/lwm2m/lwm2m_obj_firmware.c:
```diff
@@ -247,7 +247,12 @@ static int package_uri_write_cb(uint16_t obj_inst_id, uint16_t res_id,
-		lwm2m_firmware_start_transfer(package_uri);
+
+		if (data_len > 0)
+		{
```
Done.

rlubos commented on subsys/net/lib/lwm2m/lwm2m_obj_firmware.c:
```diff
@@ -247,7 +247,10 @@ static int package_uri_write_cb(uint16_t obj_inst_id, uint16_t res_id,
 		lwm2m_firmware_set_update_result(RESULT_DEFAULT);
-		lwm2m_firmware_start_transfer(package_uri);
+
+		if (data_len > 0){
```
space needed before `{`

marinjurjevic replied:
Done

rlubos reviewed (commented):
The commit body line is too long:
> 3: UC4 Line exceeds max length (126>72): "Fix for a problem in current lwm2m firmware object implementation. Transfer should not begin when an empty string is received."

rlubos reviewed (approved):
LGTM, thanks!
